console/backend/src/test/java/org/frankframework/management/web/ConfigurationStatusTest.java

```diff
@@ -0,0 +1,201 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+
+import org.frankframework.management.Action;
+import org.springframework.http.MediaType;
+import org.springframework.messaging.Message;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+@ContextConfiguration(classes = {WebTestConfiguration.class, ConfigurationStatus.class})
+public class ConfigurationStatusTest extends FrankApiTestBase {
+	public static Stream<Arguments> updateReceiverOptions() {
+		return Stream.of(
+				Arguments.of("{\"action\":\"stop\"}", Action.STOPRECEIVER),
+				Arguments.of("{\"action\":\"start\"}", Action.STARTRECEIVER),
+				Arguments.of("{\"action\":\"incthread\"}", Action.INCTHREADS),
+				Arguments.of("{\"action\":\"decthread\"}", Action.DECTHREADS)
+		);
+	}
+
+	@Test
+	public void getAdapters() throws Exception {
+		testActionAndTopicHeaders("/adapters", "ADAPTER", "GET");
+	}
+
+	@Test
+	public void getAdapter() throws Exception {
+		testActionAndTopicHeaders("/configurations/configuration/adapters/name", "ADAPTER", "FIND");
+	}
+
+	@Test
+	public void getAdapterHealth() throws Exception {
+		testActionAndTopicHeaders("/configurations/configuration/adapters/name/health", "HEALTH", null);
+	}
+
+	@Test
+	public void updateAdapters() throws Exception {
+		ArgumentCaptor<Message<Object>> requestCapture = ArgumentCaptor.forClass(Message.class);
+		doCallRealMethod().when(outputGateway).sendAsyncMessage(requestCapture.capture());
+
+		mockMvc.perform(MockMvcRequestBuilders.put("/adapters")
+						.content("{ \"action\": \"start\", \"adapters\": [\"adapter1\", \"adapter2\"] }")
+						.accept(MediaType.APPLICATION_JSON)
+						.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.status().isAccepted());
+
+		Message<Object> capturedRequest = Awaitility.await().atMost(1500, TimeUnit.MILLISECONDS).until(requestCapture::getValue, Objects::nonNull);
+		assertEquals(Action.STARTADAPTER.name(), capturedRequest.getHeaders().get("meta-action"));
+	}
+
+	@Test
+	public void updateConfigurationWithAdapters() throws Exception {
+		ArgumentCaptor<Message<Object>> requestCapture = ArgumentCaptor.forClass(Message.class);
+		doCallRealMethod().when(outputGateway).sendAsyncMessage(requestCapture.capture());
+
+		mockMvc.perform(MockMvcRequestBuilders.put("/adapters")
+						.content("{ \"action\": \"start\", \"adapters\": [\"config1/adapter1\", \"config1/adapter2\"] }")
+						.accept(MediaType.APPLICATION_JSON)
+						.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.status().isAccepted());
+
+		Message<Object> capturedRequest = Awaitility.await().atMost(1500, TimeUnit.MILLISECONDS).until(requestCapture::getValue, Objects::nonNull);
```

Comment:
Eigen moet hier ook nog iets van `verify(mock, times(2))` in.

Reply:
toegevoegd